pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -182,9 +146,16 @@ func (rc *reconciler) reconcile() {
 			// This check must be done before we do any other checks, as otherwise the other checks
 			// may pass while at the same time the volume leaves the pending state, resulting in
 			// double detach attempts
-			if rc.attacherDetacher.IsOperationPending(attachedVolume.VolumeName, "") {
-				klog.V(10).Infof("Operation for volume %q is already running. Can't start detach for %q", attachedVolume.VolumeName, attachedVolume.NodeName)
-				continue
+			if util.IsMultiAttachForbidden(attachedVolume.VolumeSpec) {
+				if rc.attacherDetacher.IsOperationPending(attachedVolume.VolumeName, "" /* podName */, "" /* nodeName */) {
```

Comment:
a small thing, change "" to nestedpendingoperations.EmptyNodeName and similar places?

Reply:
also to save some code, maybe define podName and nodeName based on util.IsMultiAttachForbidden, and then share the same code.

Reply:
>  small thing, change "" to nestedpendingoperations.EmptyNodeName and similar places?

The reconciler package actually doesn't import the nestedpendingoperations package. By referring to nestedpendingoperations.EmptyNodeName we might be breaking the abstraction boundary, though ideally Empty*Names should really be moved to a different place. On the other hand, I'm doing a refactor that will largely remove the need of specifying empty arguments, so IMO it's OK to leave them here for now.

> also to save some code, maybe define podName and nodeName based on util.IsMultiAttachForbidden, and then share the same code. Same thing applied below during attach.
The log messages are slight different between the two cases to make it more obvious which branch was followed

Reply:
it is a level 10 log, Also I don't see that difference matters. you log the volumename, nodename, isn't that enough?

Reply:
IMO it's still good to keep it because the log messages say slightly different things to more accurately describe what kind of operation is pending.